In [25]:
import pandas as pd
import numpy as np

import matplotlib.pyplot as plt

from sklearn.model_selection import train_test_split
from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import accuracy_score, classification_report

In [15]:
df = pd.read_csv('run_or_walk.csv',)
#                 low_memory=False,
                 #parse_dates=['date'],
                 #infer_datetime_format=True )

In [16]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 88588 entries, 0 to 88587
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   date            88588 non-null  object 
 1   time            88588 non-null  object 
 2   username        88588 non-null  object 
 3   wrist           88588 non-null  int64  
 4   activity        88588 non-null  int64  
 5   acceleration_x  88588 non-null  float64
 6   acceleration_y  88588 non-null  float64
 7   acceleration_z  88588 non-null  float64
 8   gyro_x          88588 non-null  float64
 9   gyro_y          88588 non-null  float64
 10  gyro_z          88588 non-null  float64
dtypes: float64(6), int64(2), object(3)
memory usage: 7.4+ MB


In [17]:
df.head().transpose()

0                   1                   2  \
date                     2017-6-30           2017-6-30           2017-6-30   
time            13:51:15:847724020  13:51:16:246945023  13:51:16:446233987   
username                    viktor              viktor              viktor   
wrist                            0                   0                   0   
activity                         0                   0                   0   
acceleration_x               0.265              0.6722              0.4399   
acceleration_y             -0.7814             -1.1233             -1.4817   
acceleration_z             -0.0076             -0.2344              0.0722   
gyro_x                      -0.059             -0.1757             -0.9105   
gyro_y                      0.0325              0.0208              0.1063   
gyro_z                     -2.9296              0.1269             -2.4367   

                                 3                   4  
date                     2017-6-30           2017-6-30  
time            13:51:16:646117985  13:51:16:846738994  
username                    viktor              viktor  
wrist                            0                   0  
activity                         0                   0  
acceleration_x              0.3031              0.4814  
acceleration_y             -0.8125             -0.9312  
acceleration_z              0.0888              0.0359  
gyro_x                      0.1199              0.0527  
gyro_y                     -0.4099              0.4379  
gyro_z                     -2.9336              2.4922

In [22]:
y = df.activity
drop_cols = ['activity', 'username', 'date', 'time']
x = df.drop(drop_cols, axis=1)

In [23]:
trainX, testX, trainY, testY = train_test_split(x, y,
                                                test_size=0.3,
                                                random_state=0)

In [27]:
model_NB = GaussianNB().fit(trainX, trainY)
y_pred = model_NB.predict(testX)
accuracy_score(testY, y_pred)

0.9567671294728525

In [29]:
print(classification_report(testY, y_pred))

              precision    recall  f1-score   support

           0       0.93      0.99      0.96     13238
           1       0.99      0.92      0.96     13339

    accuracy                           0.96     26577
   macro avg       0.96      0.96      0.96     26577
weighted avg       0.96      0.96      0.96     26577



In [31]:
trainX, testX, trainY, testY = train_test_split(
    x[['acceleration_x', 'acceleration_y', 'acceleration_z']],
    y,
    test_size=0.3,
    random_state=0
)
model_NB = GaussianNB().fit(trainX, trainY)
y_pred = model_NB.predict(testX)
accuracy_score(testY, y_pred)

0.9570681416262181

In [32]:
trainX, testX, trainY, testY = train_test_split(
    x[['gyro_x', 'gyro_y', 'gyro_z']],
    y,
    test_size=0.3,
    random_state=0
)
model_NB = GaussianNB().fit(trainX, trainY)
y_pred = model_NB.predict(testX)
accuracy_score(testY, y_pred)

0.6471008766978967

>## The model has a 95% accuracy with just the acceleration features however a model with just gyro features resulted in only ~65% accuracy